### PR TITLE
feat: remote js execution code split

### LINKED ISSUE #32318: [Feat]: Update local and redux store to handle test payload
- Update redux store to add the payload to store on startup
- Add hook to update index DB with new payload when run in triggered

### LINKED ISSUE #32320: [Feat]: Update run JS function hook to stop executeWorkflow run in browser
- Check for test run payload to be present in the indexDB
    - If empty show empty screen (adding only placeholder with this issue)
    - If not empty, trigger API call for the running on temporal

### LINKED ISSUE #32321: [Feat]: Update the response tab to show test run screen based on current running action
- If non workflow action, show normal response tab
- If workflow action, show test run view
    - If there’s no payload in the test run view, show no payload screen (add UI as per figma)
    - If there’s payload, start the run if run button is clicked

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
This PR is a code split PR for EE feature which allows remote JS execution for JS functions. The changes majorly relate to workflows code. Changelog:
- Block certain js function executions in `JsPaneSagas`
- Create store changes to allow test payload for a collection to be added to the redux store
- Save and fetch methods for test payloads in index db.
- Handle views of js response view based on local or remote execution.

Fixes #32318 
Fixes #32320
Fixes #32321 

## Automation

/ok-to-test tags="@tag.Sanity, @tag.JS"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8946098385>
> Commit: ff7aeec5c1237cb278e046f74bc66383874531e3
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8946098385&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: app/client/src/ee/components/JSRemoteExecutionView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default function JSRemoteExecutionView({}: any) {
```
Comment (ankitakinger) [inline app/client/src/ee/components/JSRemoteExecutionView/index.tsx:1]: I think you have missed the CE version of this file?

File: app/client/src/ce/sagas/JSActionSagas.ts [hunk 2]
```diff
@@ -522,3 +523,38 @@ export function* logJSFunctionExecution(
 ) {
   /* This is intentionally left blank and has a definition on EE for audit logs, since this function needs to be called in a common file. */
 }
+
+// Saga to fetch stored test payloads for all collections present in the application
+export function* fetchStoredTestPayloadsSaga(collections: JSCollection[]) {
+  try {
+    //fetch stored test payloads for all collections
+    const storedPayloads: Record<string, any> | null =
```
Comment (rishabhrathod01) [inline app/client/src/ce/sagas/JSActionSagas.ts:531]: Could we please avoid use `any` as type.

Comment (ayushpahwa) [reply app/client/src/ce/sagas/JSActionSagas.ts]: updated, please check

File: app/client/src/ee/components/JSRemoteExecutionView/index.tsx (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/ee/components/JSRemoteExecutionView/index.tsx]: Updated, please check

**Review by ankitakinger — DISMISSED** [review]
LGTM

File: app/client/src/ce/utils/actionExecutionUtils.ts [hunk 3]
```diff
@@ -83,3 +84,22 @@ export function getActionExecutionAnalytics(
...
+  return true;
+}
+
+// Function to extract the test payload from the collection data
+export const getTestPayloadFromCollectionData = (
+  collectionData: JSCollectionData | undefined,
+): string => {
+  if (!collectionData) return "";
+  const activeJSActionId = collectionData?.activeJSActionId;
+  const testPayload: Record<string, any> | undefined =
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/utils/actionExecutionUtils.ts:101]: Not really. I think the data type is currently being set to `any` somewhere in the eval pipeline. I haven't been able to track it down. So just to make things congruent, I added this.

File: app/client/src/ce/utils/actionExecutionUtils.ts [hunk 4]
```diff
@@ -83,3 +84,22 @@ export function getActionExecutionAnalytics(
 
   return resultObj;
 }
+
+// Function to check if the browser execution is allowed for the action
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/utils/actionExecutionUtils.ts:88]: Thanks for pointing this out, updated

File: app/client/src/ce/sagas/JSActionSagas.ts [hunk 5]
```diff
@@ -522,3 +523,38 @@ export function* logJSFunctionExecution(
...
+      yield getAllActionTestPayloads();
+    if (!!storedPayloads && collections.length > 0) {
+      for (const collection of collections) {
+        const testPayloadForCollection: Record<string, any> = {};
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/sagas/JSActionSagas.ts:513]: Done

**Review by rishabhrathod01 — APPROVED** [review]

**Pull request merged**